packages/backend/src/server.ts

```diff
@@ -1,5 +1,84 @@
-import app from "./app.js";
+import express from 'express';
+import cors from 'cors';
+import helmet from 'helmet';
+import { ApolloServer } from '@apollo/server';
+import { expressMiddleware } from '@apollo/server/express4';
+import { ApolloServerPluginDrainHttpServer } from '@apollo/server/plugin/drainHttpServer';
+import { applyMiddleware } from 'graphql-middleware';
+import http from 'http';
+import { logger } from '~/loggers';
+import { createContext } from '~/context';
+import { schema } from '~/schema';
+import { NODE_ENV, PORT } from '~/config';
+import permissions from '~/permissions';
+// import sentryPlugin from '~/apolloPlugins/sentry';
+import pinoLogger from '~/apolloPlugins/logger';
 
-app.listen(process.env.PORT || "8080", () => {
-  console.log("Server started");
-});
+const start = async () => {
+  const app = express();
+
+  app.use(
+    helmet({
+      contentSecurityPolicy: {
+        // these directives are required for the Apollo sandbox to work
+        directives: {
+          imgSrc: [
+            `'self'`,
+            'data:',
+            'apollo-server-landing-page.cdn.apollographql.com',
+          ],
+          scriptSrc: [`'self'`, `https: 'unsafe-inline'`],
+          manifestSrc: [
+            `'self'`,
+            'apollo-server-landing-page.cdn.apollographql.com',
+          ],
+          frameSrc: [`'self'`, 'sandbox.embed.apollographql.com'],
+        },
+      },
+    }),
+  );
+
+  app.use(
+    cors({
+      exposedHeaders: ['Authorization'],
+    }),
+  );
+
+  // health check endpoint
+  app.get('/healthz', (_req, res) => {
+    res.send('Ok');
+  });
+
+  const httpServer = http.createServer(app);
+
+  const graphqlPath = '/api/graphql';
+  const graphqlSchema = applyMiddleware(schema, permissions);
+
+  const server = new ApolloServer({
+    schema: graphqlSchema,
+    introspection: NODE_ENV !== 'production',
+    logger: logger,
+    plugins: [
+      // sentryPlugin,
```

Comment:
Is the Sentry plugin something we might want to have as a prompt option?

Reply:
Maybe, but it feels like a reasonable enough default for our backend to me.

It's also not immediately clear to me how we'd conditionally bring in all the code for this plugin... 🤔